src/main/java/de/tum/in/www1/artemis/service/scheduled/QuizScheduleService.java

```diff
@@ -409,10 +414,12 @@ private Participation createParticipationWithResultAndWriteItInHashMaps(QuizExer
             participation.setExercise(quizExercise);
 
             // save participation, result and quizSubmission
-            participationRepository.save(participation);
+            Participation storedParticipation = participationRepository.save(participation);
             quizSubmissionRepository.save(quizSubmission);
             resultRepository.save(result);
 
+            this.quizSubmissionWebsocketService.sendParticipation(storedParticipation);
```

Comment:
When the detail page was loaded for a quiz very soon after submitting, the quiz state was still UNINITIALIZED, showing the startExercise button.
The problem was that the QuizScheduler needs some time to save the participation and so it might not be there yet when the detail page is loaded.

Reply:
There was another issue I noticed: If we don't detach the results from the participation, the user in the client will know the result before the deadline passes. Therefore I set the results to null before sending out the participation.

Reply:
I had to remove this as it led to new issues.